docs/Deprecating-Disabling-and-Removing-Casks.md

```diff
@@ -0,0 +1,110 @@
+# Deprecating, Disabling and Removing Casks
+
+There are many reasons why casks may be deprecated, disabled or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+These general rules of thumb can be followed:
+
+- `deprecate!` should be used for casks that _should_ no longer be used.
+- `disable!` should be used for casks that _cannot_ be used.
+- Casks that are no longer acceptable in `homebrew/cask` or have been disabled for over a year _should_ be removed.
+
+## Deprecation
+
+If a user attempts to install a deprecated cask, they will be shown a warning message but the install will proceed.
+
+A cask should be deprecated to indicate to users that the cask should not be used and will be disabled in the future. Deprecated casks should continue to be maintained by the Homebrew maintainers if they continue to be installable. If this is not possible, they should be immediately disabled.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained, or archived.
+
+Casks should only be be deprecated if at least one of the following are true:
+
+- the cask does not install on any of our supported macOS versions
+- the cask has outstanding CVEs
+
+To deprecate a cask, add a `deprecate!` call. This call should include a deprecation date (in the ISO 8601 format) and a deprecation reason:
+
+```ruby
+deprecate! date: "YYYY-MM-DD", because: :reason
+```
+
+The `date` parameter should be set to the date that the project or version became (or will become) deprecated. If there is no clear date but the cask needs to be deprecated, use today's date. If the `date` parameter is set to a date in the future, the cask will not become deprecated until that date. This can be useful if the upstream developers have indicated a date when the project or version will stop being supported.
```

Comment:
The script I made also checks the commit date of the deprecation, so for disabling this would not happen.

Reply:
Backdating feels a bit weird/confusing but good to know the script checks commit dates.

Reply:
Agreed that we should not backdate things. The wording here states that if known we should set the deprecation date to when the project became deprecated, which is my main issue here. If we find a project we package was deprecated a few years ago, under this wording we should immediately set the deprecation date to that date, which will blow through our entire cycle straight to removal.

Reply:
Backdating really serves no meaningful purpose. In the end deprecation is a notice that we're going to remove the package and when we deliver that notice to end users is the most important thing. We're unlikely to shorten a deprecation cycle because we discovered upstream archived their repo a few months ago.

Even though Sean's script might check commit dates, many others do not and there's been multiple incidents of early disabling as a result - I last saw one as recent as last month.

Reply:
100% with you. It only causes problems, and should be avoided.